manager/handlers/user.go

```diff
@@ -48,3 +48,75 @@ func (h *Handlers) SignUp(ctx *gin.Context) {
 
 	ctx.JSON(http.StatusOK, user)
 }
+
+// @Summary Revoke Role
+// @Description Revoke Role by uri config
+// @Tags users
+// @Accept json
+// @Produce json
+// @Success 200
+// @Failure 400 {object} HTTPError
+// @Failure 500 {object} HTTPError
+// @Router /users/:user_name/roles/:role_name [delete]
+
+func (h *Handlers) RevokeRole(ctx *gin.Context) {
+	var params types.RoleRequest
+	if err := ctx.ShouldBindUri(&params); err != nil {
+		ctx.JSON(http.StatusUnprocessableEntity, gin.H{"errors": err.Error()})
+		return
+	}
+	err := h.Service.RevokeRole(params.UserNmae, params.RoleName)
+	if err != nil {
+		ctx.Error(err)
+		return
+	}
+	ctx.Status(http.StatusOK)
+}
+
+// @Summary Grant Role
+// @Description Grant Role by uri config
+// @Tags users
+// @Accept json
+// @Produce json
+// @Success 200
+// @Failure 400 {object} HTTPError
+// @Failure 500 {object} HTTPError
+// @Router /users/:user_name/roles/:role_name [post]
+
+func (h *Handlers) GrantRole(ctx *gin.Context) {
+	var params types.RoleRequest
+	if err := ctx.ShouldBindUri(&params); err != nil {
+		ctx.JSON(http.StatusUnprocessableEntity, gin.H{"errors": err.Error()})
+		return
+	}
+	err := h.Service.GrantRole(params.UserNmae, params.RoleName)
+	if err != nil {
+		ctx.Error(err)
+		return
+	}
+	ctx.Status(http.StatusOK)
+}
+
+// @Summary Get User Roles
+// @Description Get User Roles
+// @Tags User
+// @Produce json
+// @Success 200 {object} RoutesInfo
+// @Failure 400 {object} HTTPError
+// @Failure 500 {object} HTTPError
+// @Router /user/:user_name/roles [get]
+
+func (h *Handlers) GetRolesForUser(ctx *gin.Context) {
+	var params types.UserParams
+	if err := ctx.ShouldBindUri(&params); err != nil {
+		ctx.JSON(http.StatusUnprocessableEntity, gin.H{"errors": err.Error()})
+		return
+	}
+	roles, err := h.Service.GetRolesForUser(params.UserName, ctx.GetString("userName"))
```

Comment:
Add `Pagination`

Reply:
The result return by casbin, it dou't support pagination.